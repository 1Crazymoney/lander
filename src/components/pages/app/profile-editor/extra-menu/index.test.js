import React from 'react';
import ExtraMenu from './index';
import renderer from 'react-test-renderer';


it('1- No data provided. should shown', () => {
  const props = {
    toggleUiProp: () => {
    },
    draft: {
      wallets: {
        bitcoin: '',
        ethereum: ''
      },
      contact: {
        email: ''
      }
    }
  };

  const component = <ExtraMenu {...props} />;

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


it('2- Wallet data and contact data provided. should render null ', () => {
  const props = {
    toggleUiProp: () => {
    },
    draft: {
      wallets: {
        bitcoin: '21123213',
        ethereum: 'x01312321'
      },
      contact: {
        email: 'foo@bar'
      }
    }
  };

  const component = <ExtraMenu {...props} />;

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


it('3- Contact provided but wallets. should shown', () => {
  const props = {
    toggleUiProp: () => {
    },
    draft: {
      wallets: {
        bitcoin: '',
        ethereum: ''
      },
      contact: {
        email: 'foo@bar.com'
      }
    }
  };

  const component = <ExtraMenu {...props} />;

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


it('4- undefined contact data. for backward compability. should shown', () => {
  const props = {
    toggleUiProp: () => {
    },
    draft: {
      wallets: {
        bitcoin: 'x1x1',
        ethereum: ''
      }
    }
  };

  const component = <ExtraMenu {...props} />;

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
