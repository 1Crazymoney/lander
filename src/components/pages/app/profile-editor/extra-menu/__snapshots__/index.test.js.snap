// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- No data provided. should shown 1`] = `
<div
  className="extra-menu with-2-items"
>
  <div
    className="dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-link btn-sm"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      <svg
        data-icon="ellipsis-h"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`2- Wallet data and contact data provided. should render null  1`] = `null`;

exports[`3- Contact provided but wallets. should shown 1`] = `
<div
  className="extra-menu with-1-items"
>
  <div
    className="dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-link btn-sm"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      <svg
        data-icon="ellipsis-h"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`4- undefined contact data. for backward compability. should shown 1`] = `
<div
  className="extra-menu with-1-items"
>
  <div
    className="dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-link btn-sm"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      <svg
        data-icon="ellipsis-h"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;
